anago

```diff
@@ -449,18 +449,24 @@ EOF
    || return 1
 
   # Sync $CHANGELOG_FILE to release-* branch
-  if [[ "$RELEASE_BRANCH" =~ release- ]]; then
-    logecho -n "Checkout $RELEASE_BRANCH branch to make changes: "
-    logrun -s git checkout $RELEASE_BRANCH || return 1
-    logecho -n "Remove any previous CHANGELOG-*.md files: "
-    logrun -s git rm -f CHANGELOG-*.md || return 1
-    logecho -n "Copy master $CHANGELOG_FILE to $RELEASE_BRANCH branch: "
-    logrun -s git checkout master -- $CHANGELOG_FILE || return 1
-    logecho -n "Committing $CHANGELOG_FILE: "
-    logrun -s git commit -am \
-              "Add/Update $CHANGELOG_FILE for $RELEASE_VERSION_PRIME." \
-     || return 1
-  fi
+  [[ "$RELEASE_BRANCH" =~ release- ]] && clear_sync_changelogs \
+   || return 1
+}
+
+##############################################################################
+# Clear all other (not current) branch changelogs and sync the master
+# CHANGELOG_FILE to the branch
+clear_sync_changelogs () {
+  logecho -n "Checkout $RELEASE_BRANCH branch to make changes: "
+  logrun -s git checkout $RELEASE_BRANCH || return 1
```

Comment:
I'm a little worried that this branch switch is inside a function. You can't tell at the call site that this function will change the meaning of future git commands.

Can we either make the caller responsible for checking out the release branch, or save/restore the previous branch in the function?